include/ghex/transport_layer/ucx/context.hpp

```diff
@@ -233,14 +233,19 @@ namespace gridtools {
             template<class ThreadPrimitives>
             struct context_factory<ucx_tag, ThreadPrimitives>
             {
-                static std::unique_ptr<context<ucx_tag, ThreadPrimitives>> create(int num_threads, MPI_Comm mpi_comm = MPI_COMM_NULL)
+                static std::unique_ptr<context<ucx_tag, ThreadPrimitives>> create(int num_threads, MPI_Comm mpi_comm)
                 {
 #ifdef GHEX_USE_PMI
                     return std::make_unique<context<ucx_tag,ThreadPrimitives>>(num_threads, mpi_comm, ucx::address_db_pmi{mpi_comm});
 #else
                     return std::make_unique<context<ucx_tag,ThreadPrimitives>>(num_threads, mpi_comm, ucx::address_db_mpi{mpi_comm});
 #endif
                 }
+                template<typename DB>
```

Comment:
@boeschf Changed, as we discussed in the original PR https://github.com/boeschf/GHEX/pull/3#discussion_r359734426. Please have a look if this is OK.

Reply:
does it work above when passing an MPI comm and creating a pmi database (when `GHEX_USE_PMI` is enabled)?

Reply:
@boeschf yes, but at this moment the mpi comm is null in this code path - because this has to be called before MPI_Init. So the code will work, but for the barrier to work the comm needs to be updated later to a correct value. Do we want a set function for the mpi comm? 

Right now I've modified the benchmarks to use the other code path.

Reply:
I think we should not use it at all, to be safe (I don't like MPI == null). The second function which takes a database may be modified to take only a db and get the mpi-comm from it (that would eliminate a potential user error) - it would require to update the database classes, though. There could even be a default database argument (the mpi db). Then the first function would not be needed anymore.

Reply:
I don't think we can avoid `MPI_COMM_NULL`. I need to construct the `transport_context` before  `MPI_Init` is called. And in your implementation `transport_context` takes the `parallel_context` as construction argument, which requires mpi comm to be passed. This dependency has to be broken, and it can be done using `MPI_COMM_NULL`, or dropping the `parallel_context` from `transport_context` constructor.